back-end/routes/api/tables.js

```diff
@@ -51,23 +51,35 @@ router.post("/submit", async (req, res) => {
     state: req.body.state,
     zipcode: req.body.zipcode,
     locationWithinBuilding: req.body.locationWithinBuilding,
-    restroomType: req.body.restroomType,
-    tableStyle: req.body.tableStyle,
     tableNotes: req.body.tableNotes,
-    publiclyAccessible: req.body.publiclyAccessible,
     hours: req.body.hours,
     contactPhone: req.body.contactPhone,
     contactEmail: req.body.contactEmail,
     additionalInfo: req.body.additionalInfo,
     status: 'submitted'
   });
 
-  newTable.save()
-    .then(table => res.json({
+  if (req.body.restroomType) {
+    newTable.restroomType = req.body.restroomType;
+  }
+
+  if (req.body.tableStyle) {
+    newTable.tableStyle = req.body.tableStyle
+  }
+
+  if (req.body.publiclyAccessible) {
+    newTable.publiclyAccessible = req.body.publiclyAccessible
+  }
+
+  try {
+    const table = await newTable.save();
+    res.json({
       "success": true,
-      "data": table
-    }))
-    .catch(err => res.status(500).json({error: err}));
+      "tableId": table._id
```

Comment:
sending just the table id back instead of the whole table - front-end will use that to create a link to the View Table page for the new table